src/pynwb/tests/test_constructors.py

```diff
@@ -40,17 +40,34 @@ def test_constructor_excitation_light_path_failing():
 
     excitation_wavelength_in_nm = 600.0
     excitation_source = mock_ExcitationSource(excitation_wavelength_in_nm=488.0)
-    try:
+    with pytest.raises(ValueError) as e:
         _ = mock_ExcitationLightPath(
             excitation_wavelength_in_nm=excitation_wavelength_in_nm, excitation_source=excitation_source
         )
-    except ValueError as e:
-        assert str(e) == (
-            f"wavelength set in the light path ({excitation_wavelength_in_nm}) and the one set in the device "
-            f"({excitation_source.excitation_wavelength_in_nm}) must be the same."
-        )
-    else:
-        assert False, "Expected ValueError not raised"
+    assert str(e.value) == (
```

Comment:
rather than asserting this manually you can use pytest.raises match argument. Ex.
```python
expected_error_message = (
    f"wavelength set in the light path ({excitation_wavelength_in_nm}) and the one set in the device "
    f"({excitation_source.excitation_wavelength_in_nm}) must be the same."
)
with pytest.raises(ValueError, match=expected_error_message)
```